tests/qnn/test_keras.py

```diff
@@ -542,3 +542,35 @@ def test_model_save_weights(self, model, n_qubits, tmpdir):
         assert np.allclose(prediction, prediction_loaded)
         for i, w in enumerate(weights):
             assert np.allclose(w, weights_loaded[i])
+
+    @pytest.mark.parametrize("n_qubits, output_dim", indices_up_to(1))
+    def test_model_serialization(self, get_circuit, output_dim):
+        """Test if serializing and deserializing `KerasLayer` as part of a 
+        Keras model reconstructs the layer correctly, i.e. that the right
+        arguments are passed to __init__ when instantiating the layer during
+        deserialization."""
```

Comment:
Suggest rephrasing more as model cloning than serialization, since that is the main function being used.

Reply:
Agreed. I'll change it in the next commit.